apiv2/internal/server/host.go

```diff
@@ -719,54 +706,84 @@ func (is *InventorygRPCServer) GetHostsSummary(
 	var errorState uint32
 	var runningState uint32
 	var unallocatedState uint32
+	reqFilter := req.GetFilter()
+
+	filterIsFailedHostStatus := `%s = %s OR %s = %s OR %s = %s OR %s.%s = %s OR %s.%s = %s OR %s.%s = %s OR %s.%s = %s`
+	filterIsFailedHostStatus = fmt.Sprintf(filterIsFailedHostStatus,
+		inv_computev1.HostResourceFieldHostStatusIndicator,
+		api.HostResourceHostStatusIndicatorSTATUSINDICATIONERROR,
+		inv_computev1.HostResourceFieldOnboardingStatusIndicator,
+		api.HostResourceHostStatusIndicatorSTATUSINDICATIONERROR,
+		inv_computev1.HostResourceFieldRegistrationStatusIndicator,
+		api.HostResourceHostStatusIndicatorSTATUSINDICATIONERROR,
+		inv_computev1.HostResourceEdgeInstance,
+		inv_computev1.InstanceResourceFieldInstanceStatusIndicator,
+		api.HostResourceHostStatusIndicatorSTATUSINDICATIONERROR,
+		inv_computev1.HostResourceEdgeInstance,
+		inv_computev1.InstanceResourceFieldProvisioningStatusIndicator,
+		api.InstanceResourceInstanceStatusIndicatorSTATUSINDICATIONERROR,
+		inv_computev1.HostResourceEdgeInstance,
+		inv_computev1.InstanceResourceFieldUpdateStatusIndicator,
+		api.InstanceResourceInstanceStatusIndicatorSTATUSINDICATIONERROR,
+		inv_computev1.HostResourceEdgeInstance,
+		inv_computev1.InstanceResourceFieldTrustedAttestationStatusIndicator,
+		api.InstanceResourceInstanceStatusIndicatorSTATUSINDICATIONERROR,
+	)
+	filterInstanceRunning := `%s.%s = %q AND NOT (%s)`
+	filterInstanceRunning = fmt.Sprintf(filterInstanceRunning,
+		inv_computev1.HostResourceEdgeInstance,
+		inv_computev1.InstanceResourceFieldInstanceStatus,
+		"Running",
+		filterIsFailedHostStatus,
+	)
 
-	isFailedHostStatus := func(host *computev1.HostResource) bool {
-		hostErr := host.GetHostStatusIndicator() == statusv1.StatusIndication_STATUS_INDICATION_ERROR ||
-			host.GetOnboardingStatusIndicator() == statusv1.StatusIndication_STATUS_INDICATION_ERROR
-
-		instanceErr := false
-		if host.GetInstance() != nil {
-			instanceErr = host.GetInstance().
-				GetInstanceStatusIndicator() ==
-				statusv1.StatusIndication_STATUS_INDICATION_ERROR ||
-				host.GetInstance().GetProvisioningStatusIndicator() == statusv1.StatusIndication_STATUS_INDICATION_ERROR ||
-				host.GetInstance().GetUpdateStatusIndicator() == statusv1.StatusIndication_STATUS_INDICATION_ERROR
-		}
-		return hostErr || instanceErr
+	filterIsUnallocated := `NOT has(%s) OR %s.%s = %q`
+	filterIsUnallocated = fmt.Sprintf(filterIsUnallocated,
+		inv_computev1.HostResourceEdgeSite,
+		inv_computev1.HostResourceEdgeSite,
+		inv_locationv1.SiteResourceFieldResourceId,
+		"")
```

Comment:
Do we really need this second part of the filter? Can it ever be empty the ResourceID? I'm not sure why we had such a filter in the past.

Reply:
Fixed.